### PR TITLE
New ngmix_fit submodule 

### LINKED ISSUE #26: Add general ngmix fitting module
It can be hard to use metacalibration photometry for validation as we know Gaussians are not a good fit to the underlying galaxy profiles. It would be useful to have another pipeline module that fits standard ngmix models (e.g. exp, model, bdf) that can be run either before or after mcal. As long as we use SOF it should be fairly fast in comparison to mcal.

### PR DESCRIPTION
**Pull request by sweverett**
See #26. Is now fully integrated w/ the pipeline framework, and comes with test config generation. Only fits BFD profiles for now but simple to add others (upgrading to `ngmix` v2.x.x would eliminate this need). Long-term we can think about merging the mcal fitting code with this submodule. Closes #26

### CODE REVIEW

File: superbit_lensing/pipe.py [hunk 1]
```diff
@@ -107,7 +108,7 @@ def _run_command(self, cmd, logprint):
         Run bash command
         '''
 
-        logprint(cmd)
+        logprint(f'\n{cmd}\n')
```
Comment (mcclearyj) [inline superbit_lensing/pipe.py:111]: I like this!

File: superbit_lensing/galsim/config_files/pipe_test.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ center_dec:         33.1        # Central Declination                       [deg
 sky_bkg:            0.039       # mean sky background from AG's paper       [ADU / s / px]
 sky_sigma:          0.06        # standard deviation of sky background      [ADU / s / px]
 nobj:               0           # Number of galaxies (COSMOS 25.2 depth)    []
-nstars:             1000        # Number of stars in the field              []
+nstars:             10        # Number of stars in the field              []
```
Comment (mcclearyj) [inline superbit_lensing/galsim/config_files/pipe_test.yaml:28]: Dropping the number of stars to 10 for speed makes sense

**Comment by mcclearyj** [comment]
One thing I haven't been able to figure out: it seems like the `utils.py` file has changed quite a bit. Will this break the way other modules within `superbit_lensing` work? We should confirm this.

**Comment by sweverett** [comment]
The main change in `utils.py` is just that I fixed a bug in setting the `test` directory, which should be at the repo root (`superbit-metacal`) and not inside the module directory `superbit_lensing`.

I think the real problem right now is related to #29. I was working on a different project this weekend and I noticed that importing `import {repo}.utils as utils` instead of `import utils` caused the utils helper functions for getting the base dir, module dir, and test dir to evaluate to the binaries in the miniconda directories instead of the local repo directory.

I initially changed these so that you could run the test scripts in a different directory. But this behavior is unexpected. I'll try to fix it now.

**Comment by sweverett** [comment]
Hmm, maybe not. I just did a very quick test in ipython in the `superbit_lensing` directory:
```
import utils

print(utils.TEST_DIR)
> '/Users/sweveret/repos/superbit-metacal/tests'

import superbit_lensing.utils as utils_two

print(utils_two.TEST_DIR)
> '/Users/sweveret/repos/superbit-metacal/tests'
```
These were not equivalent in the other repo I was working on. I wonder if it is due to a difference in the `setup.py` files. In any case, #29 might not be a problem for us.

**Comment by sweverett** [comment]
> One thing I haven't been able to figure out: it seems like the `utils.py` file has changed quite a bit. Will this break the way other modules within `superbit_lensing` work? We should confirm this.

Also - one of the nice things about the new pipeline setup is that it is very easy to test this. `python pipe.py` will run the test config with a very quick mock simulation so that it is easy to ensure that the rest of the pipeline modules still run successfully.

**Comment by mcclearyj** [comment]
Brilliant, OK

**Pull request merged**